git_machete/cmd.py

```diff
@@ -67,30 +41,46 @@ class MacheteClient:
     def __init__(self, cli_opts: CommandLineOptions, git: GitContext) -> None:
         self.cli_opts: CommandLineOptions = cli_opts
         self.git: GitContext = git
-        self.definition_file_path: str = self.git.get_git_subpath("machete")
-        self.managed_branches: List[str] = []
+        self._definition_file_path: str = self.git.get_git_subpath("machete")
+        self._managed_branches: List[str] = []
         self.down_branches: Dict[str, List[str]] = {}  # TODO (#110): default dict with []
         self.up_branch: Dict[str, str] = {}  # TODO (#110): default dict with None
-        self.indent: Optional[str] = None
-        self.roots: List[str] = []
-        self.annotations: Dict[str, str] = {}
-        self.empty_line_status: Optional[bool] = None
+        self.__indent: Optional[str] = None
+        self.__roots: List[str] = []
+        self.__annotations: Dict[str, str] = {}
+        self.__empty_line_status: Optional[bool] = None
+
+    @property
+    def definition_file_path(self) -> str:
+        return self._definition_file_path
+
+    @definition_file_path.setter
+    def definition_file_path(self, val: str) -> None:
+        self._definition_file_path = val
+
+    @property
+    def managed_branches(self) -> List[str]:
+        return self._managed_branches
+
+    @managed_branches.setter
+    def managed_branches(self, val: List[str]) -> None:
```

Comment:
Same here, do we need to expose the setter publicly?

Reply:
Since managed_branches fields is used also outside the class, i would stick to current solution to emulate protected field. Another option would be to make it private and make helper method to return it outside.